### PR TITLE
maintainers.rst: Add a stepping down section

### PR DESCRIPTION
**Pull request by Andersson007**
maintainers.rst: Add a stepping down section

### CODE REVIEW

File: maintaining.rst [hunk 1]
```diff
@@ -307,3 +307,19 @@ Inclusion in Ansible
...
+Maintainers should not step down silently. This is especially important when the collection has one or few active maintainer.
+
+If you feel you don't have time to maintain your collection any more or for a long period of time, to prevent negative consequences for the collection and its community:
+
+  - Inform other maintainers about it.
+  - If the collection is under the ``ansible-collections`` organization, also inform the community team in the ``ansible-community`` `Libera.Chat IRC channel <https://docs.ansible.com/ansible/devel/community/communication.html#irc-channels>`_ or by email to ``ansible-community-team@redhat.com``.
+  - Look at active contributors in the collection to find new maintainers among them. Discuss the potential candidates with other maintainers or with the community team.
+  - If you failed to find a replacement, create a pinned issue in the collection which announces that the collection needs new maintainers.
+  - Make the same announcement through the `Bullhorn newsletter <https://github.com/ansible/community/wiki/News#the-bullhorn>`_.
+  - Be around to discuss potential candidates found by other maintainers or by the community team.
```
Comment (felixfontein) [inline maintaining.rst:323]:
```suggestion
  - Please be around to discuss potential candidates found by other maintainers or by the community team.
```

Comment (Andersson007) [reply maintaining.rst]: yeah, makes sense to put it:)

File: maintaining.rst [hunk 2]
```diff
@@ -307,3 +307,19 @@ Inclusion in Ansible
...
+
+Stepping down
+=============
+
+Maintainers should not step down silently. This is especially important when the collection has one or few active maintainer.
+
+If you feel you don't have time to maintain your collection any more or for a long period of time, to prevent negative consequences for the collection and its community:
+
+  - Inform other maintainers about it.
+  - If the collection is under the ``ansible-collections`` organization, also inform the community team in the ``ansible-community`` `Libera.Chat IRC channel <https://docs.ansible.com/ansible/devel/community/communication.html#irc-channels>`_ or by email to ``ansible-community-team@redhat.com``.
```
Comment (Andersson007) [inline maintaining.rst:319]:
```suggestion
  - If the collection is under the ``ansible-collections`` organization, also inform the community team in the ``ansible-community`` `Libera.Chat IRC channel <https://docs.ansible.com/ansible/devel/community/communication.html#irc-channels>`_.
```

Comment (Andersson007) [reply maintaining.rst]: as it's not a public one. Will add later

**Comment by Andersson007** [comment]
@felixfontein thanks for reviewing!

**Pull request merged**